src/lib/Simplify.hs

```diff
@@ -151,6 +149,12 @@ simplifyExpr expr = confuseGHC >>= \_ -> case expr of
             liftM Var $ emit $ Case e' alts' resultTy' eff'
           False -> defuncCase e' alts resultTy'
 
+caseComputingEffs
+  :: forall m n. (MonadFail1 m, EnvReader m)
+  => Atom n -> [Alt n] -> Type n -> m n (Expr n)
+caseComputingEffs scrut alts resultTy = do
+  Case scrut alts resultTy <$> foldMapM effectsE alts
```

Comment:
We should make sure that this either gets inline or is specialized to `SimplifyM`!

Reply:
Done (commit 289e8ceb79ae06547dcf6616176b229b1a0bf901).